docs/user/08-10-best-practices.md

```diff
@@ -0,0 +1,61 @@
+
+Initialize a scaffold for a brand new Function using the `kyma init function` command or fetch the current state of an existing Function deployed in your Kyma runtime using `kyma sync function`.
+Focus on the Function code and develop it from your favorite IDE. Configure your Functions directly in the [`config.yaml` manifest file](/docs/user/07-60-function-configuration-file.md)
+>**TIP:** Use `kyma init function --vscode` to generate json schema which can be used in VSCode for autocompletion.
+
+Kyma CLI helps you run your code locally with a single `kyma run function` command. You can run your Function using your local Docker daemon with the same runtime Docker context, as if it was run in Kyma runtime.
+
+>**TIP:** Use `kyma run function` with `--hot-deploy` and spare yourself unnecessary restarts of the Functions whenever you test a changed Function logic. Also, use [`--debug` option](/docs/user/01-40-debug-function.md) to allow connecting with your favorite debugger.
+>>
+
+![kyma-cli-functions](../assets/svls-kyma-cli-functions.png)
+
+Having written and tested your Function locally, simply deploy it to the Kyma runtime with the `kyma apply function` command, used in the folder of your Function's workspace. The command reads the files, translates them to the Kubernetes manifests, and deploys the Function.
+
+
+# Deploy using CI/CD
+
+Kyma UI helps you get started. Kyma CLI helps you iterate and develop Functions. 
+But at the end of the day, you may want an automated deployment of your application, where Functions are just part of it.
+It all comes down to the deployment the Kubernetes applications on different Kyma runtimes in a GitOps fashion. For the sake of simplicity, the deployment approach for Functions should not differ from the deployment of the other Kubernetes workloads, ConfigMaps, or Secrets.
+
+So, in the end, what you need is YAML manifests for everything - including Functions.
+
+Fortunately, Kyma CLI helps you generate the YAML mainfests matching your `config.yaml` file crafted before.
+Use the `--dry-run` option of the `kyma apply function` command to generate Kubernetes manifests that will include the Function CR itself and all the related CRs (for example, APIRules, Subscriptions, etc.).
+
+   ```bash
+   kyma apply function --dry-run --ci -o yaml > my-function.yaml
+   ```  
+
+The generated manifest should be a part of all the manifests that define your application and pushed to the git repository.
+Deploy everything in a consistent way either using CI/CD or GitOps operators (e.g. `fluxcd` or `argocd`) installed on your Kyma runtime.
+
+>**NOTE:** Kyma Functions come in two types: `git` and `inline`. The [Git type](/docs/user/01-11-create-git-function.md) allows you to configure a Git repository as a source of your Function code instead of creating it `inline`.
```

Comment:
```suggestion
>**NOTE:** Kyma Functions come in two types: `git` and `inline`. For the [Git type](/docs/user/01-11-create-git-function.md), you configure a Git repository as a source of your Function code instead of creating it `inline`.
```